services/CurrentTime/source/CurrentTimeService.cpp

```diff
@@ -68,12 +104,19 @@ void CurrentTimeService::onDataRead(GattReadAuthCallbackParams *read_request) {
     read_request->authorizationReply = AUTH_CALLBACK_REPLY_SUCCESS;
```

Comment:
before you set the data and the AUTH_CALLBACK_REPLY_SUCCESS check `current_time_is_valid()` and if it's not send an error instead